### PR TITLE
MOB-10665 Expose CSV export API in HAProxy. Add documentation and cod…

### PR DESCRIPTION
**Pull request by dkkplk**
…e sample

added sample code to download csv file

### CODE REVIEW

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java [hunk 1]
```diff
@@ -197,21 +241,63 @@ private static void downloadTestReport(Path testPdfPath, CreatePdfTask task, Str
...
+        while (updatedTask.getStatus() != TaskStatus.COMPLETE && startTime + maxGenerationTime > System.currentTimeMillis());
+
+        if (updatedTask.getStatus() == TaskStatus.COMPLETE) {
+            downloadPdfOrCsvFileToFS(testCsvPath, new URI(updatedTask.getUrl()));
+        } else {
+            throw new RuntimeException("The task is still in " + updatedTask.getStatus() + " status after waiting " + maxWaitMin + " min");
+        }
+    }
+
+    private static CreateTask getUpdatedTask(String taskId) throws URISyntaxException, IOException {
```
Comment (yuragr) [inline Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java:277]: this method and the following method that you added do basically the same thing. The only difference is that the URL is different. This means that you can create 1 method that receives the URL as a param. Also, create global constants for `REPORTING_SERVER_URL + "/export/api/v2/test-executions/pdf/task/"` and `REPORTING_SERVER_URL + "/export/api/v2/test-executions/csv/"`

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java [hunk 2]
```diff
@@ -148,12 +146,58 @@ public static void downloadTestReport(Path testPdfPath, String testId) throws UR
...
     }
 
+    /**
+     * Downloads a CSV report
+     *
+     * @param testCsvPath local path that the downloaded report will be saved to
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    public static void downloadCsvTestReport(Path testCsvPath) throws URISyntaxException, IOException {
```
[the remark this answers is not in the document]
Comment (dkkplk) [reply Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java:162]: updated the code

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java [hunk 3]
```diff
@@ -279,16 +365,35 @@ private static void addDefaultRequestHeaders(HttpRequestBase request) {
         request.addHeader("PERFECTO_AUTHORIZATION", SECURITY_TOKEN);
     }
 
+
+    private static void addRequestBody(HttpPost request) {
+        JsonObject json = new JsonObject();
+        JsonObject filerJson = new JsonObject();
+        JsonArray values = new JsonArray();
+        values.add("1641839400000");
```
[the remark this answers is not in the document]
Comment (dkkplk) [reply Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java:373]: Updated the code

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/CsvDownloadCodeSample.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static void main(String[] args) throws Exception {
+        Path tempDir = Files.createTempDirectory("reporting_csv_sample_");
+
+        // TODO put your required file name here
+        String csvFileName = "Report_Library_Sample";
+
+        // Download a CSV report - create a "task" for CSV generation and download the CSV on task completion
+        Path testCsvPath = Paths.get(tempDir.toString(), csvFileName + ".csv");
+        ReportiumExportUtils.downloadCsvTestReport(testCsvPath);
```
[the remark this answers is not in the document]
Comment (dkkplk) [reply Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/CsvDownloadCodeSample.java:22]: added the required code

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java [hunk 5]
```diff
@@ -148,12 +146,58 @@ public static void downloadTestReport(Path testPdfPath, String testId) throws UR
...
+                int statusCode = response.getStatusLine().getStatusCode();
+                if (HttpStatus.SC_OK == statusCode) {
+                    task = gson.fromJson(EntityUtils.toString(response.getEntity()), CreateTask.class);
+                    break;
+                } else if (HttpStatus.SC_NO_CONTENT == statusCode) {
+
+                    // if the execution is being processed, the server will respond with empty response and status code 204
+                    System.out.println("The server responded with 204 (no content). " +
+                            "The execution is still being processed. Attempting again in 5 sec (" + attempt + "/" + DOWNLOAD_ATTEMPTS + ")");
+                    Thread.sleep(5000);
```
[the remark this answers is not in the document]
Comment (dkkplk) [reply Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java:183]: removed the unwanted code

File: Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java (hunk 1 above)
Comment (dkkplk) [reply Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java:277]: updated the code

**Review by yuragr — APPROVED** [review]

**Pull request merged**